src/algos/input_optimization.py

```diff
@@ -0,0 +1,116 @@
+import json
+from utils.config_utils import process_config
+from algos.uniform_noise import UniformNoise
+from algos.nopeek import NoPeek
+from algos.simba_algo import SimbaAttack
+from utils.metrics import MetricLoader
+import torch
+from torchvision.utils import save_image
+
+class InputOptimization(SimbaAttack):
+  def __init__(self, config, utils):
+    super().__init__(utils)
+    self.initialize(config)
+
+  def initialize(self, config):
+    self.attribute = config["attribute"]
+    self.model_name = config["target_model"]
+
+    # load obfuscator model
+    target_exp_config = json.load(open(config["target_model_config"])) #config_loader(config["model_config"])
+    system_config = json.load(open("./configs/system_config.json")) #config_loader(config["model_config"])
+    target_config = process_config(system_config, target_exp_config)
+    self.target_config = target_config
+
+    if self.model_name == "uniform_noise":
```

Comment:
https://github.com/splitlearning/InferenceBenchmark/blob/main/src/interface.py#L39 use this one